CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -188,12 +192,22 @@ local function AddTool(tool)
 
 	slot.hovered = false
 	slot.tool = tool
-	slot.icon = Instance.new("ImageButton", toolsFrame)
+	if tool.TextureId ~= "" then
```

Comment:
Instead of assigning two types to the same variable, lets add a image and text property.  It might make sense to allow for both someday, and it stops the need for a string check above